### ISSUE TITLE
Sidecar requires RestTemplate bean

### ISSUE THREAD

**Issue by marcingrzejszczak** [spring-cloud/spring-cloud-netflix#3239]
For `Finchley.BUILD-SNAPSHOT` Sidecar requires a bean of type `RestTemplate` to be explicitly set

```
java.lang.IllegalStateException: Failed to load ApplicationContext
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'healthCheckHandler' defined in org.springframework.cloud.netflix.sidecar.SidecarConfiguration$EurekaInstanceConfigBeanConfiguration: Unsatisfied dependency expressed through method 'healthCheckHandler' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'localApplicationHealthIndicator': Unsatisfied dependency expressed through field 'restTemplate'; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.springframework.web.client.RestTemplate' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true)}
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'localApplicationHealthIndicator': Unsatisfied dependency expressed through field 'restTemplate'; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.springframework.web.client.RestTemplate' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true)}
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.springframework.web.client.RestTemplate' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true)}
```

You can check https://github.com/spring-cloud/spring-cloud-core-tests `Finchley` branch.

The failing tests are

```
demo.SidecarApplicationTests.instanceConfigHasCorrectPort
demo.SidecarApplicationTests.contextLoads
```

**Comment by ryanjbaxter** [comment]
It was this that broke it
https://github.com/spring-cloud/spring-cloud-netflix/commit/53e7464a6ed5b115b22ef7618e99acc68be47cd5

**Comment by ryanjbaxter** [comment]
Closed via 1897c95
